meshtastic/tests/seturl/test_rollback_engine.py

```
@@ -0,0 +1,439 @@
+
+        admin_context = MagicMock()
+        admin_context.admin_index_for_write = 0
+
+        failed_admin_indexes: list[int] = []
+
+        def write_side_effect(
+            _channel: channel_pb2.Channel, *, adminIndex: int
+        ) -> None:
+            if adminIndex == 0:
+                failed_admin_indexes.append(adminIndex)
+                raise _WriteFailure("Write failed")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)